src/mango/src/mango_httpd.erl

```diff
@@ -266,6 +267,11 @@ run_find(Resp, Db, Sel, Opts) ->
     mango_crud:find(Db, Sel, fun handle_doc/2, Acc0, Opts).
 
 
+handle_doc({add_key, Key, Value = #execution_stats{}}, Acc0) ->
```

Comment:
I think what Tony is saying here is you don't need to do the `lists:keystore` in this function. Rather convert to json and then call `handle_doc()` again so that the function below will do all lists:keystore work.
I also think you could consider doing the `to_json` and in the cursor:execute sections. Then you don't need to add a whole new function.